### ISSUE TITLE
It would be nice to have nupnp config option again for privacy reasons

### ISSUE THREAD

**Issue by bachandi** [ebaauw/homebridge-hue#887]
The nupnp option was removed in the latest release but it would still be nice to have the option to force homebridge to work without internet access and the need to leave traces at philips/hue via discovery.meethue.com.
Could the nupnp config option be reintroduced?

**Comment by ebaauw** [comment]
Homebridge Hue works fine without an Internet connection.  You might see a couple of timeout messages in the log that in cannot reach the Meethue and Phoscon portals  (and the NPM registry), but it should work otherwise.  

I fail to see the privacy issue of querying these portals.  Unlike registering the bridge or gateway, querying doesn’t (visibly) store any data on the portal.  I don’t know about Signify or dresden elektronik, but I cannot see whether you queried the portal, even if we would share a public IP address.  Anyway, you can block access to these portals in your firewall, without breaking Homebridge Hue.

**Comment by bachandi** [comment]
But with the nupnp config one could tell homebridge-hue from the beginning to not even try to query and not have to wait for timeout. If you have a fixed setup there is no need for any internet traffic.

It would be nice to have this option to reduce internet traffic if not needed and not have to block it via firewall. Like for TVs and other smart things it is nice to have a setting to just disable calling home and not have to block it.

By accessing discovery.meethue.com the users IPs could be tracked by Signify or the Server hoster like Amazon or Google. They could build a profile, match it with other information they have on your IP like access to an Amazon/Google account via this IP. Not saying they do it but it would be possible. So an explicit switch to reduce internet traffic would be nice if there is no need to have this traffic. For some services you can't avoid it but for non essential stuff it is always nice to have at least the option to minimise data traces.

But by specifying `hosts` you loose the local discovery right? Haven't looked in the code for some while but think to remember there was discovery still possible locally without internet query?

**Comment by ebaauw** [comment]
> But with the nupnp config one could tell homebridge-hue from the beginning to not even try to query and not have to wait for timeout.

That's a non-issue: querying the portals happens in parallel with local discovery over UPnP, using the same timeout (as set in config.json).

> It would be nice to have this option to reduce internet traffic

That's a non-issue: the payload of the request and the reponse is very small.

> By accessing discovery.meethue.com the users IPs could be tracked by Signify or the Server hoster like Amazon or Google. They could build a profile, match it with other information they have on your IP like access to an Amazon/Google account via this IP. Not saying they do it but it would be possible.

Fair enough.  The same argument would hold for the NPM registry (to check for the latest version).

> For some services you can't avoid it but for non essential stuff it is always nice to have at least the option to minimise data traces.

OK, I'll have a look.  It wouldn't be `nupnp`, but a privacy setting to prevent any calls to servers on the Internet (incl. the NPM registry).

> But by specifying `hosts` you loose the local discovery right? Haven't looked in the code for some while but think to remember there was discovery still possible locally without internet query?

Yes, you'd also skip local discovery through UPnP.

**Comment by bachandi** [comment]
> OK, I'll have a look. It wouldn't be `nupnp`, but a privacy setting to prevent any calls to servers on the Internet (incl. the NPM registry).

That would be great.

**Comment by ebaauw** [comment]
Added `stealth` in v0.12.15.
